### PR TITLE
INIT STC

### PR DESCRIPTION
**Pull request by sakshibobade21**
## Proposed changes

This PR addresses Issue: The PR incorporates STC initialization to ZEN
This PR depends upon the following PRs:

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Change in a documentation
- [ ] Refactor the code 
- [ ] Chore, repository cleanup, updates the dependencies.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## PR Checklist
Please delete options that are not relevant.
- [ ] If the changes in this PR are meant for the next release / mainline, this PR targets a "staging" branch.
- [ ] My code follows the style guidelines of this project (see: [Contributing guideline](https://github.com/zowe/zen/blob/v2.x/staging/CONTRIBUTING.md))
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] video or image is included if visual changes are made

- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, or describe a test method below

### CODE REVIEW

File: src/renderer/preload.ts [hunk 1]
```diff
@@ -115,18 +115,24 @@ contextBridge.exposeInMainWorld('electron', {
...
+    getInitStcsProgress(){
+      return ipcRenderer.invoke("get-init-stcs-progress");
+    },
     getInitVsamProgress(){
       return ipcRenderer.invoke("get-init-vsam-progress");
     },
     on(channel: string, func: any) {
       // REVIEW: Used to have channel validation with ipcRenderer.send, do we need something similar for ipcRenderer.invoke?
-      const validChannels = ['install-mvs', 'init-security', 'init-vsam'];
+      const validChannels = ['install-mvs', 'init-security', 'init-vsam', 'init-stcs'];
```
Comment (1000TurquoisePogs) [inline src/renderer/preload.ts]: comment here on why this doesnt include certificates or apf stages? intentional?

File: src/renderer/components/stages/Stcs.tsx [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+      </Box>
+      <ContainerCard title="Stcs" description="Install Zowe Main Started Tasks.">
+        {editorVisible && <EditorDialog contentType={contentType} isEditorVisible={editorVisible} toggleEditorVisibility={toggleEditorVisibility} onChange={handleFormChange}/> }
+        <Typography id="position-2" sx={{ mb: 1, whiteSpace: 'pre-wrap', color: 'text.secondary', fontSize: '13px' }}>
+          {`Please review the following started task (STC) configuration values from the security stage before initializing stcs.\n`}
+        </Typography>
+        <Box sx={{ width: '60vw' }} onBlur={async () => dispatch(setYaml((await window.electron.ipcRenderer.getConfig()).details.config ?? yaml))}>
+          {!isFormValid && <div style={{color: 'red', fontSize: 'small', marginBottom: '20px'}}>{formError}</div>}
+          
+          <Box sx={{ width: '60vw' }}>
```
Comment (1000TurquoisePogs) [inline src/renderer/components/stages/Stcs.tsx:258]: where is zowe.setup.dataset.proclib?

these 3 members go into the proclib, so it must be shown somewhere

**Review by 1000TurquoisePogs — COMMENTED** [review]
I can't comment much on the code, seems mostly boilerplate from other steps which is fine, but please ensure this PR accomplishes the following:


* "init stc" copes 3 members into the proclib

That is,

zowe.setup.security.stcs.zowe
zowe.setup.security.stcs.zis
zowe.setup.security.stcs.aux
all go into

zowe.setup.dataset.proclib

* There is no other "zwe init" step which uses "zowe.setup.dataset.proclib", so it should be visible and customizable.

* Each of the "zowe.setup.security.stcs" entries are also used in "zwe init security", so they should be editable in one step and read-only in the other step.

File: src/renderer/components/stages/Stcs.tsx (hunk 2 above)
Comment (sakshibobade21) [reply src/renderer/components/stages/Stcs.tsx:258]: Good point. I will add that.

File: src/renderer/preload.ts (hunk 1 above)
Comment (sakshibobade21) [reply src/renderer/preload.ts]: Valid channels are allowed for listening to events. Must also contain apf auth & certificate. Thanks

**Review by DivergentEuropeans — APPROVED** [review]
This was tested by QA. I didn't test it fully but we don't have a lot of time, so we need to test staging in bulk

**Pull request merged**